src/openslide-vendor-aperio.c

```diff
@@ -558,6 +572,12 @@ static bool aperio_open(openslide_t *osr,
     } else {
       // associated image
       const char *name = (dir == 1) ? "thumbnail" : NULL;
+      if (gt450_workaround) {
+          if(dir == tiff_dir_count-2)
+              name = "label";
+          if (dir == tiff_dir_count-1)
+              name = "macro";
```

Comment:
There's a comment in https://github.com/openslide/openslide/issues/297 that you should check TIFFTAG_SUBFILETYPE instead of doing this. This will mislabel frames in files that don't have a label or macro frame (they're optional after all).